### PR TITLE
feat: drop django 2.2, 3.0, and 3.1 support

### PR DESCRIPTION
**Pull request by dopry**
Drop support for EOL Django versions 2.2, 3.0, and 3.1

## Motivation and Context
As an open-source team, we don't have the resources to maintain compatibility with end-of-life code.  Consumers who need to work with older versions of Django can use older releases. 

## Checklist:
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: .coveragerc
Comment (moggers87) [inline .coveragerc:21]: This config line appears to have caused CodeCov to report missing coverage on `tests/__init__.py`. We either need to configure CodeCov to also skip empty files or remove this line and put up with empty files in our coverage reports.

**Review by moggers87 — CHANGES_REQUESTED** [review]
Looks good, just one changed needed to fix coverage issues.

**Comment by claudep** [comment]
As a release might come soon, I would suggest to wait a bit and merge this just after the release.

File: .coveragerc
Comment (dopry) [reply .coveragerc]: Here is coverage without skip-empty
```
@@            Coverage Diff             @@
##           master     #497      +/-   ##
==========================================
- Coverage   98.42%   98.34%   -0.08%     
==========================================
  Files          55       58       +3     
  Lines        2475     2606     +131     
  Branches      266      274       +8     
==========================================
+ Hits         2436     2563     +127     
- Misses         24       26       +2     
- Partials       15       17       +2     
```

The diff is 100% covered, I feel like the -0.08% is an error in  the change in coverage caldulation in this case.

**Comment by claudep** [comment]
I don't trust codecov results for a long time now... don't bother.

File: .coveragerc
Comment (claudep) [inline .coveragerc:19]: Avoid unrelated changes in this patch, please.

File: CHANGELOG.md
Comment (claudep) [inline CHANGELOG.md:16]: Please move this above (in a new Unreleased section), please.

File: two_factor/plugins/yubikey/__init__.py
Comment (claudep) [inline two_factor/plugins/yubikey/__init__.py:4]: I think there are more django.VERSION checks in master.

**Review by claudep — CHANGES_REQUESTED** [review]
Now that 1.14.0 is out, we can go forward with this.

File: CHANGELOG.md
Comment (dopry) [reply CHANGELOG.md]: fixed.  dealing with CHANGELOG conflicts isn't very contributor friendly. It would be nice if these were a part of the release process, either manually aggregated by the maintainers or by something like [semantic-release](https://python-semantic-release.readthedocs.io/en/latest/).

File: two_factor/plugins/yubikey/__init__.py
Comment (claudep) [reply two_factor/plugins/yubikey/__init__.py]: It may be that the comparison is wrong and should have been `< (3, 2)`. Could you check?

**Review by claudep — APPROVED** [review]
Thanks!

**Comment by claudep** [comment]
@moggers87 We need your review to go forward as you requested changes.

**Review by moggers87 — APPROVED** [review]

**Pull request merged**
